spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -86,13 +94,22 @@ private void validateAndSetFilterParts() {
 					"Cloud Datastore structured queries do not support the Distinct keyword.");
 		}
 
-		List parts = this.tree.getParts().get().collect(Collectors.toList());
+		if ((!this.tree.getSort().equals(Sort.unsorted())) && (getQueryMethod().getParameters().hasPageableParameter()
+				|| getQueryMethod().getParameters().hasSortParameter())) {
+			throw new DatastoreDataException("Ambiguous sort options (method name and parameter).");
```

Comment:
Is there a guard against the user providing more than one pageable or sort parameter?

Reply:
I don't think it's our responsibility to check that. Instead, `Parameters` class should deal with that.